forte/mrdsrg-spin-integrated/dsrg_mrpt2.h

```diff
@@ -176,6 +179,259 @@ class DSRG_MRPT2 : public MASTER_DSRG {
+     */
+    void solve_z();
+    /**
+     * Initialize and solve the multiplier Sigma and Xi.
+     *
+     * Sigma: constraint of the one-body DSRG amplitude (T1) definition.
+     * Solved directly.
+     */
+    void set_sigma_xi();
+    /**
+     * Initialize and solve the multiplier Tau.
+     *
+     * Tau: constraint of the two-body DSRG amplitude (T2) definition.
+     * Solved directly.
+     */
+    void set_tau();
+    /**
+     * Initialize and solve the multiplier Kappa.
+     *
+     * Kappa: constraint of the renormalized ERIs (\tilde{V}) definition.
+     * Solved directly.
+     */
+    void set_kappa();
+    /**
+     * Initialize and solve the multiplier Z (OPDM).
+     *
+     * Z: OPDM, constraint of the CASSCF reference.
+     * The core-core, virtual-virtual blocks and diagonal entries of the active-active
+     * blocks are solved directly based on other multipliers. The rest need be solved
+     * Through iterative approaches. Currently, we use LAPACK as the solver.
+     */
+    void set_z();
+    /**
+     * Setting the b of the Linear System Ax=b.
+     * Parameters: preidx, block_dim
+     */
+    void set_b(int, std::map<string, int>, std::map<string, int>);
+    /**
+     * The diagonal core-core, virtual-virtual blocks 
+     * and the diagonal entries of the active-active block of the OPDM Z.
+     */
+    void set_z_diag();
+    /**
+     * Initialize and solve the multiplier W.
+     *
+     * W: EWDM, constraint of the orthonormal overlap integral.
+     * Solved directly after all other multipliers are solved.
+     */
+    void set_w();
+    // Set MO orbital partition info for solving the z-vector equation
+    void set_zvec_moinfo();
+
+    /// Size of determinants
+    size_t ndets;
+    /// List of core MOs (Correlated)
+    std::vector<size_t> core_mos;
+    /// List of active MOs (Correlated)
+    std::vector<size_t> actv_mos;
+    /// List of virtual MOs (Correlated)
+    std::vector<size_t> virt_mos;
+    /// List of core MOs (Absolute)
+
+    std::vector<size_t> core_all;
+    /// List of active MOs (Absolute)
+    std::vector<size_t> actv_all;
+    std::vector<size_t> virt_all;
+
+    
+    // MO orbital partition info for solving the z-vector equation
+    int dim;
+    std::map<string, int> preidx;
+    std::map<string, int> block_dim;
+
+    /// List of relative core MOs
+    std::vector<std::pair<unsigned long, unsigned long>,
+                std::allocator<std::pair<unsigned long, unsigned long>>>
+        core_mos_relative;
+    /// List of relative active MOs
+    std::vector<std::pair<unsigned long, unsigned long>,
+                std::allocator<std::pair<unsigned long, unsigned long>>>
+        actv_mos_relative;
+    /// List of relative virtual MOs
+    std::vector<std::pair<unsigned long, unsigned long>,
+                std::allocator<std::pair<unsigned long, unsigned long>>>
+        virt_mos_relative;
+
+    /// Dimension of different irreps
```

Comment:
Just to check. Do all of these have to be class members or could we have some defined locally in the functions where they are generated/used?